addons/mail/sfu_server/src/services/http.js

```diff
@@ -0,0 +1,249 @@
+import https from "node:https";
+
+import * as ws from "#src/services/ws.js";
+import * as auth from "#src/services/auth.js";
+import * as config from "#src/config.js";
+import * as rtc from "#src/services/rtc.js";
+import { Logger, parseBody, extractRemoteAddress } from "#src/utils/utils.js";
+import { SESSION_CLOSE_CODE } from "#src/models/session.js";
+import { Channel } from "#src/models/channel.js";
+
+/**
+ * @typedef {function} routeCallback
+ * @param {http.IncomingMessage} req
+ * @param {http.ServerResponse} res
+ * @param {Object} param2
+ * @param {string} param2.remoteAddress
+ * @return {http.ServerResponse}
+ */
+
+export const API_VERSION = 1;
+const logger = new Logger("HTTP");
+let httpServer;
+/** @type{Boolean} */
+let isSSL = config.PROXY === "SSL";
+
+/**
+ * @param {Object} [options]
+ * @param {string} [options.hostname]
+ * @param {number} [options.port]
+ */
+export async function start({ hostname = config.HOSTNAME, port = config.PORT } = {}) {
+    logger.info("starting...");
+    const routeListener = new RouteListener();
+    routeListener.get(`/v${API_VERSION}/noop`, {
+        callback: (req, res) => {
+            res.statusCode = 200;
+            return res.end(JSON.stringify({ result: "ok" }));
+        },
+    });
+    routeListener.get(`/v${API_VERSION}/stats`, {
+        callback: async (req, res, { remoteAddress }) => {
+            res.setHeader("Content-Type", "application/json");
+            req.statusCode = 200;
+            const heapTotal = process.memoryUsage().heapTotal;
+            const loadAvg = os.loadavg();
+            logger.info(
+                `[${remoteAddress}] fetched stats: last minute load average: ${loadAvg[0]}, total memory: ${heapTotal}`
+            );
+            return res.end(
+                JSON.stringify({
+                    workerStats: await rtc.getWorkersResourceUsages(),
+                    heapTotal,
+                    loadAvg,
+                })
+            );
+        },
+    });
+    routeListener.get(`/v${API_VERSION}/channel`, {
+        callback: async (req, res, { remoteAddress }) => {
+            try {
+                const jsonWebToken = req.headers.authorization?.split(" ")[1];
+                /** @type {{ iss: string }} */
+                const claims = await auth.verify(jsonWebToken);
+                if (!claims.iss) {
+                    logger.warn(`${remoteAddress}: missing issuer claim when creating channel`);
+                    res.statusCode = 403; // forbidden
+                    return res.end();
+                }
+                const channel = await Channel.create(remoteAddress, claims.iss);
+                res.setHeader("Content-Type", "application/json");
+                res.statusCode = 200;
+                return res.end(
+                    JSON.stringify({
+                        uuid: channel.uuid,
+                        url: `${isSSL ? "https://" : "http://"}${config.PUBLIC_IP}:${port}`,
```

Comment:
ok, got it

Reply:
talking about this address, maybe the port there should be a different env variable? `port` is currently based on the `PORT` env variable but if it's behind a proxy it could be a different port no?